sunpy/timeseries/timeseriesbase.py

```diff
@@ -544,8 +546,17 @@ def truncate(self, a, b=None, int=None):
             start = a
             end = b
 
+        min_time, max_time = self._data.index.min(), self._data.index.max()
+        # Check if the timerange overlaps with the data timerange
+        if not (isinstance(a ,int) and isinstance(b , int)):
```

Comment:
`def truncate(self, a, b=None, int=None):`

`def truncate(self, a, b=None, step=None):`

in this line we are using int class that's why we can't you int as variable within function

Reply:
Yes I understand but we can't just break the API. We have to go through a depreciation cycle. 

https://docs.sunpy.org/en/latest/dev_guide/contents/public_api.html#deprecation-policy-and-breaking-changes

Reply:
I think it is just better to change the logic